### ISSUE TITLE
Build libtorch failed

### ISSUE THREAD

**Issue by eric-tc** [pytorch/pytorch#50289]
## To Reproduce

I'am building libtorch 1.6 following this guide https://github.com/pytorch/pytorch/blob/master/docs/libtorch.rst. 
I'm using Cmake

I'm using this https://github.com/pytorch/pytorch/issues/33124#issuecomment-602048845 to solve qnnpack compilation problem with GCC

/home/ubuntu/Develop/Installer/pytorch/torch/csrc/distributed/rpc/utils.cpp: In function ‘std::tuple<tensorpipe::Message, torch::distributed::rpc::TensorpipeWriteBuffers> torch::distributed::rpc::tensorpipeSerialize(torch::distributed::rpc::Message&&)’:
/home/ubuntu/Develop/Installer/pytorch/torch/csrc/distributed/rpc/utils.cpp:513:55: error: could not convert ‘storageData.std::vector<char>::data()’ from ‘char*’ to ‘tensorpipe::Buffer’
           tensorpipe::Message::Tensor{storageData.data(), storageData.size()});
                                       ~~~~~~~~~~~~~~~~^~
/home/ubuntu/Develop/Installer/pytorch/torch/csrc/distributed/rpc/utils.cpp:513:75: error: could not convert ‘storageData.std::vector<char>::size()’ from ‘std::vector<char>::size_type {aka long unsigned int}’ to ‘std::__cxx11::string {aka std::__cxx11::basic_string<char>}’
           tensorpipe::Message::Tensor{storageData.data(), storageData.size()});
                                                           ~~~~~~~~~~~~~~~~^~
/home/ubuntu/Develop/Installer/pytorch/torch/csrc/distributed/rpc/utils.cpp:521:74: error: could not convert ‘tensorPtr’ from ‘char*’ to ‘tensorpipe::Buffer’
           tensorpipe::Message::Tensor{tensorPtr, tensorData.sizeInBytes()});
                                                                          ^
/home/ubuntu/Develop/Installer/pytorch/torch/csrc/distributed/rpc/utils.cpp:521:72: error: could not convert ‘(& tensorData)->torch::jit::WriteableTensorData::sizeInBytes()’ from ‘size_t {aka long unsigned int}’ to ‘std::__cxx11::string {aka std::__cxx11::basic_string<char>}’
           tensorpipe::Message::Tensor{tensorPtr, tensorData.sizeInBytes()});
                                                  ~~~~~~~~~~~~~~~~~~~~~~^~
/home/ubuntu/Develop/Installer/pytorch/torch/csrc/distributed/rpc/utils.cpp: In function ‘torch::distributed::rpc::TensorpipeReadBuffers torch::distributed::rpc::tensorpipeAllocate(tensorpipe::Message&)’:
/home/ubuntu/Develop/Installer/pytorch/torch/csrc/distributed/rpc/utils.cpp:566:70: error: ‘struct tensorpipe::Message::Tensor’ has no member named ‘length’
     buffers.tensors.push_back(at::getCPUAllocator()->allocate(tensor.length));
                                                                      ^~~~~~
/home/ubuntu/Develop/Installer/pytorch/torch/csrc/distributed/rpc/utils.cpp:567:12: error: ‘struct tensorpipe::Message::Tensor’ has no member named ‘data’; did you mean ‘metadata’?
     tensor.data = buffers.tensors.back().get();
            ^~~~
            metadata
[ 67%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/api/src/nn/modules/activation.cpp.o
[ 67%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/api/src/nn/modules/adaptive.cpp.o
[ 67%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/api/src/nn/modules/batchnorm.cpp.o
caffe2/CMakeFiles/torch_cpu.dir/build.make:23777: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/distributed/rpc/utils.cpp.o' failed
make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/distributed/rpc/utils.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:3021: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed
make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2


## Expected behavior



## Environment

 - Libtorch Version : 1.6
 - OS :Linux ubuntu-xavier 4.9.140-tegra
 - How you installed PyTorch (`conda`, `pip`, source): Cmake
 - Build command you used (if compiling from source): cmake -DBUILD_SHARED_LIBS:BOOL=ON -DUSE_NCCL=OFF -DCMAKE_BUILD_TYPE:STRING=Release -DUSE_CUDA:BOOL=OFF -DPYTHON_EXECUTABLE:PATH=`which python3` -DUSE_CUDNN:BOOL=OFF -DCMAKE_INSTALL_PREFIX:PATH=../pytorch-install ../pytorch
 - Python version: 3.6.9
 - CUDA/cuDNN version: 10.2/8.0.0.190
 - Compiler gcc (Ubuntu/Linaro 7.5.0-3ubuntu1~18.04) 7.5.0



## Additional context

**Comment by skyline75489** [comment]
Please do not delete what is in the Issue Template and provide the necessary context accordingly. For example the compiler version.

**Comment by eric-tc** [comment]
HI @skyline75489,
ok i have updated the comment.

Thanks

**Comment by skyline75489** [comment]
Have you tried submodule update? Seems a submodule issue. Also cleaning the build folders sometimes help.

**Comment by eric-tc** [comment]
HI @skyline75489 

thanks for the suggestion. Right now i have compiled it by adding 

`-DUSE_DISTRIBUTED:BOOL=OFF `

to Cmake command. 
I will try also by updating submodule and let you know.

**Comment by eric-tc** [comment]
Hi @skyline75489 ,

i was able to compile following your suggestion.

Thanks

**Comment by HyeonseopLim** [comment]
@eric-tc  @skyline75489 Hi, I have a same error. What do you mean updating submodule? apt upgrade?
